airflow/utils/python_virtualenv.py

```diff
@@ -76,11 +88,18 @@ def prepare_virtualenv(
         See virtualenv documentation for more information.
     :param requirements: List of additional python packages.
     :param requirements_file_path: Path to the ``requirements.txt`` file.
+    :param pip_install_options: a list of pip install options when installing requirements
+        See 'pip install -h' for available options
+    :param index_urls: an optional list of index urls to load Python packages from.
+        If not provided the system pip conf will be used to source packages from.
     :return: Path to a binary file with Python in a virtual environment.
     """
     if pip_install_options is None:
         pip_install_options = []
 
+    if index_urls is not None:
+        _generate_pip_conf(Path(venv_directory) / "pip.conf", index_urls)
```

Comment:
Any reason a config file is preferred over passing everytrhing in the pip command below?

Reply:
Yes. When adding the pip config into the command line there is a risk of leaking credentials in logs. If credentials are in a conf then it is not exposed to stdout. (Passwords might be looking different when URL encoded and then standard secret masking might miss this)

Reply:
Sounds reasonable, thanks. I wonder if this would conflict with custom virtualenv creators/seeders, but I guess those things are pretty seldom used (too seldom if you ask me) so we can deal with them when they pop up.